### PR TITLE
[Utopia 1291] check PI PIA to make sure preselect adm role in program area if condition satisify

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- see title
Screenshot 2023-07-14 at 4 21 49 PM

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Review by Bottle7 — APPROVED** [review]

File: src/frontend/src/components/public/PIAFormTabs/review/index.tsx [hunk 1]
```diff
@@ -69,6 +78,47 @@ const PIAReview = ({ printPreview }: IReviewProps) => {
...
+        ?.storedOutsideCanada === YesNoInput.YES &&
+      pia?.storingPersonalInformation?.sensitivePersonalInformation
+        .doesInvolve === YesNoInput.YES &&
+      pia?.storingPersonalInformation?.sensitivePersonalInformation
+        .disclosedOutsideCanada === YesNoInput.NO
+    ) {
+      setMandatoryADM(true);
+      if (
+        !pia?.review?.programArea.selectedRoles.includes(ApprovalRoles.ADM) &&
+        !reviewForm.programArea?.selectedRoles.includes(ApprovalRoles.ADM)
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/review/index.tsx:104]: why checking for both? 🤔

Comment (binggaofw) [reply src/frontend/src/components/public/PIAFormTabs/review/index.tsx:104]: !reviewForm.programArea?.selectedRoles.includes(ApprovalRoles.ADM) will move the second check

File: src/frontend/src/components/public/PIAFormTabs/review/index.tsx [hunk 2]
```diff
@@ -69,6 +78,44 @@ const PIAReview = ({ printPreview }: IReviewProps) => {
...
+  useEffect(() => {
+    // if the condition does satisfy the rule, add the adm to programArea
+    // otherwise do nothing
+    if (
+      pia?.storingPersonalInformation?.personalInformation
+        ?.storedOutsideCanada === YesNoInput.YES &&
+      pia?.storingPersonalInformation?.sensitivePersonalInformation
+        .doesInvolve === YesNoInput.YES &&
+      pia?.storingPersonalInformation?.sensitivePersonalInformation
+        .disclosedOutsideCanada === YesNoInput.NO
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/review/index.tsx:99]: Are we sure this has to NO?

Comment (binggaofw) [reply src/frontend/src/components/public/PIAFormTabs/review/index.tsx:99]: yes, 
Screenshot 2023-07-18 at 12 05 13 PM

**Comment by binggaofw** [comment]
Screenshot 2023-07-18 at 2 06 17 PM

**Review by kushal-arora-fw — APPROVED** [review]

**Review by antsand — APPROVED** [review]

**Pull request merged**